### PR TITLE
Fixes JAWR-383: Provide Spring Boot auto configuration

### PR DESCRIPTION
**Pull request by dtrunk90**
I accidentally removed the HelloController :-/

### CODE REVIEW

File: jawr-spring/jawr-spring-boot-sample/pom.xml [hunk 1]
```diff
@@ -63,13 +62,17 @@
...
 		</dependency>
 		<!-- end::actuator[] -->
 		<dependency>
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-webmvc</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>com.github.dtrunk90</groupId>
+			<artifactId>thymeleaf-jawr-extension</artifactId>
+		</dependency>
```
Comment (ic3fox) [inline jawr-spring/jawr-spring-boot-sample/pom.xml]: Hi Danny

Is the dependency to thymelea-jawr-extension required for this feature?
Would it be possible to remove?

Cheers

Comment (dtrunk90) [inline jawr-spring/jawr-spring-boot-sample/pom.xml]: I guess you could add `<scope>provided</scope>` here. The auto configuration adds a Thymeleaf Dialect if the jar is in the projects classpath: `@ConditionalOnClass(name = "com.github.dtrunk90.thymeleaf.jawr.dialect.JawrDialect")`

Comment (ic3fox) [inline jawr-spring/jawr-spring-boot-sample/pom.xml]: I would like to remove circular dependency if possible.

Would it be possible to remove the JawrDialect reference in JawrAutoConfiguration and provide a custom JawrThymeleafAutoConfiguration in the **thymeleaf-jawr-extension** ?

Comment (dtrunk90) [inline jawr-spring/jawr-spring-boot-sample/pom.xml]: Sure.

**Comment by dtrunk90** [comment]
Now it should be good to go

**Pull request merged**